pallets/liquidity-pools/src/lib.rs

```diff
@@ -689,6 +692,26 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::weight(10_000)]
+		#[pallet::call_index(10)]
+		pub fn schedule_upgrade(origin: OriginFor<T>, contract: DomainAddress) -> DispatchResult {
```

Comment:
Note that `Message::ScheduleUpgrade` holds a `contract: Address` where `Address` is a 32 byte array. In light of what I said above, I am of the opinion we should make this a 20-byte array, since we don't have any plans to use this message for other domains and don't even know if 32 bytes would work for that hypotetical new use case. TL;DR make it specific for EVM since that's what this message is here to do.

If we don't, then the code suggestion above needs to be adjusted to: `Message::ScheduleUpgrade { contract: vec_to_fixed_array(contract.to_vec()) },`

cc @offerijns

Reply:
I think making it a 20 byte array makes sense. If we later want to support other contract types, since this is the last message argument, that would anyways be a non-breaking change :)

Reply:
Ah right, didn't think about that :) thanks 👌